libp2p/connection.nim

```diff
@@ -25,7 +25,7 @@ const
   ConnectionTrackerName* = "libp2p.connection"
 
 type
-  Connection* = ref object of LPStream
+  Connection* {.acyclic.} = ref object of LPStream
```

Comment:
I actually never used that pragma, does it make a difference? Wonder if it even works :)

Reply:
The compiler itself heavily depends on this pragma. It will run much slower if it doesn't work.

Reply:
But this should compiler heuristics, we should not be marking cycles. Marking might lead to big leaks too..

Reply:
In fact Connection IS NOT acyclic and might leak in this case

Reply:
I wonder if this works as well, I've added it temporarily to see if there is any change in behavior. The reason is the recent `readLoops` that could (as per the comment) introduce cycles, but I'm not sure if placing the pragma on the full object will lead to the expected behavior and it doesn't seem to be applicable at the param level...

Reply:
Yes, you should definitely not use this pragma if cycles are possible. This may lead to memory leaks. The compiler is not trying to analyze whether the usage is safe in any way, it's up to you guarantee that.